### PR TITLE
231039 : Publish Library chart ADP-Helm-Repository

### PR DESCRIPTION
**Pull request by ganeshg2491**
- ADO Pipeline to package and push library chart to 'adp-helm-repository'

AB#232259

### CODE REVIEW

File: .azuredevops/publish-helm-library.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: $(BuildID)
+
+trigger:
+  batch: true
+  branches:
+    include:
+      - 'main'
+  paths:
+    include:
+      - .azuredevops/*
```
Comment (ppotturi) [inline .azuredevops/publish-helm-library.yaml]: I think you can put * as you include everything.

File: .azuredevops/publish-helm-library.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      - adp-helm-library/*
+      - scripts/*
+
+pr:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - .azuredevops/*
```
Comment (ppotturi) [inline .azuredevops/publish-helm-library.yaml]: same here. * ?

File: scripts/Publish-HelmLibrary.ps1 [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:HelmLibraryPath=$HelmLibraryPath"
+Write-Debug "${functionName}:ChartVersion=$ChartVersion"
+Write-Debug "${functionName}:HelmChartRepoPublic=$HelmChartRepoPublic"
+
+try {
+    Write-Host "Package Helm library chart"
+    helm package $HelmLibraryPath
```
Comment (ppotturi) [inline scripts/Publish-HelmLibrary.ps1]: There are many CLI commands in the script. i think we should use the "Command-Invoke" module or check the LASTERRORCODE for all the commands.

File: .azuredevops/publish-helm-library.yaml (hunk 1 above)
Comment (ganeshg2491) [reply .azuredevops/publish-helm-library.yaml]: @ppotturi : Actually I have not included files outside above mentioned folders.

File: .azuredevops/publish-helm-library.yaml (hunk 2 above)
Comment (ganeshg2491) [reply .azuredevops/publish-helm-library.yaml]: @ppotturi : I have not included files outside above mentioned folders.

File: scripts/Publish-HelmLibrary.ps1 (hunk 3 above)
Comment (ganeshg2491) [reply scripts/Publish-HelmLibrary.ps1]: @ppotturi  Added LASTERRORCODE check

**Review by aadev1 — DISMISSED** [review]

File: scripts/Publish-HelmLibrary.ps1 (hunk 3 above)
Comment (ppotturi) [reply scripts/Publish-HelmLibrary.ps1]: There are more CLI's I think use the ps helper module to make it consistent with the other scripts.

File: .azuredevops/publish-helm-library.yaml (hunk 1 above)
Comment (ganeshg2491) [reply .azuredevops/publish-helm-library.yaml]: @ppotturi : fixed

File: scripts/Publish-HelmLibrary.ps1 (hunk 3 above)
Comment (ganeshg2491) [reply scripts/Publish-HelmLibrary.ps1]: @ppotturi : fixed

**Comment by ppotturi** [comment]
Can you please update the title with Story number: {workitem}:{title}

**Review by ppotturi — APPROVED** [review]

**Review by aadev1 — APPROVED** [review]

**Pull request merged**